vaccine_feed_ingest/stages/enrichment.py

```diff
@@ -234,6 +258,105 @@ def _add_provider_tag(loc: location.NormalizedLocation) -> None:
     ]
 
 
+def _bulk_geocode(
+    locs: Collection[location.NormalizedLocation],
+    geocodio_api: Optional[GeocodioAPI],
+) -> None:
+    """Geocode and fix addreses if missing"""
+    if geocodio_api is None:
+        return
+
+    def _full_address(loc: location.NormalizedLocation) -> Optional[str]:
+        # Only process if something is missig
+        if loc.location and _valid_address(loc):
+            return None
+
+        combined_address = []
+        if loc.address.street1:
+            combined_address.append(loc.address.street1)
+        if loc.address.street2:
+            combined_address.append(loc.address.street2)
+        if loc.address.city:
+            combined_address.append(loc.address.city)
+        if loc.address.state:
+            combined_address.append(loc.address.state)
+        if loc.address.zip:
+            combined_address.append(loc.address.zip)
+
+        return ", ".join(combined_address)
+
+    records = {}
+    for loc in locs:
+        full_address = _full_address(loc)
+        if full_address:
+            records[loc.id] = full_address
+
+    places = geocodio_api.batch_geocode(records)
+
+    if not places:
+        return
+
+    for loc in locs:
+        place_results = places.get(loc.id)
+
+        if not place_results:
+            continue
+
+        # Only trust the result if exactly one is returned
+        if len(place_results) != 1:
+            continue
```

Comment:
With the site I tested locally, we geocoded 6 locations, but 4 had more than one result. This may be a good place to re-evaluate in the future!

Should we log when we're skipping for this so we can gauge?

Reply:
Added logging